pkg/ingester/ingester.go

```diff
@@ -1833,13 +1842,42 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
 	queue := make(chan string)
 	group, groupCtx := errgroup.WithContext(ctx)
 
+	userIDs, err := i.getAllUsersWithTSDB()
+	if err != nil {
+		level.Error(i.logger).Log("msg", "error while finding existing TSDBs", "err", err)
+		return err
+	}
+
+	if len(userIDs) == 0 {
+		return nil
+	}
+
+	var concurrentOpenTSDBCount = i.cfg.BlocksStorageConfig.TSDB.DeprecatedMaxTSDBOpeningConcurrencyOnStartup
+	var walReplayConcurrency = 0
+	// If TSDBConfig.WALReplayConcurrency is set to a positive value, we honor it and ignore value of
+	// TSDB.DeprecatedMaxTSDBOpeningConcurrencyOnStartup, being the latter deprecated.
+	// If TSDBConfig.WALReplayConcurrency is 0, it is ignored, and TSDB.DeprecatedMaxTSDBOpeningConcurrencyOnStartup
+	// determines the number of concurrent processes opening TSDBs.
+	if i.cfg.BlocksStorageConfig.TSDB.WALReplayConcurrency > 0 {
```

Comment:
It might be nice to pull this logic to a function to clean up the code here https://github.com/grafana/mimir/pull/4445/files#diff-e1032332627c413a3010c66b54b22b6e9835cf152fa339e40cf0b11204f7241fR1855-R1872

Reply:
This will be cleaned up when the deprecated configuration property `TSDB.DeprecatedMaxTSDBOpeningConcurrencyOnStartup` gets removed. So for the moment I would leave this code here unless you have strong preferences to move it out from here.